solara/lab/components/input_date.py

```diff
@@ -0,0 +1,190 @@
+    label: str = "Select dates",
+    children: List[solara.Element] = [],
+    open: Union[solara.Reactive[bool], bool] = False,
+    date_format: str = "%Y/%m/%d",
+    first_day_of_the_week: int = 1,
+    style: Optional[Union[str, Dict[str, str]]] = None,
+):
+    """
+    Show a textfield, which when clicked, opens a datepicker that allows users to select a range of dates by choosing a starting and ending date.
+    The input dates should be stored in a reactive list of type `datetime.date`. The list should contain either precisely two elements.
+    For an empty pre-selection of dates, pass a reactive empty list.
+
+    ## Basic Example
+
+    ```solara
+    import solara
+    import solara.lab as lab
+
+
+    @solara.component
+    def Page():
+        dates = solara.use_reactive([])
+        range_is_open = solara.use_reactive(False)
+
+        controls = [
+            solara.Button(
+                label="Book",
+                color="primary",
+                on_click=lambda: range_is_open.set(not range_is_open.value),
+            )
+        ]
+
+        with solara.Column():
+            lab.InputDateRange(
+                dates,
+                children=[
+                    solara.Row(children=controls, justify="end", style="width: 100%;")
+                ],
+                open=range_is_open,
+            )
+    ```
+
+    ## Arguments
+
+    * value: Reactive tuple with elements of type `datetime.date`. For an empty pre-selection of dates, pass a reactive empty tuple.
+    * label: Text used to label the text field that triggers the datepicker.
+    * children: List of Elements to be rendered under the calendar. If empty, a close button is rendered.
+    * open: Controls and communicates the state of the datepicker. If True, the datepicker is open. If False, the datepicker is closed.
+    Intended to be used in conjunction with a custom set of controls to close the datepicker.
+    * date_format: Sets the format of the date displayed in the text field. Defaults to `"%Y/%m/%d"`.
+    * first_day_of_the_week: Sets the first day of the week, as an `int` starting count from Sunday (`=0`). Defaults to `1`, which is Monday.
+    """
+    date_strings = [date.strftime(date_format) if date is not None else "" for date in value.value]
+    date_standard_strings = [date.strftime("%Y-%m-%d") if date is not None else None for date in value.value]
+    datepicker_is_open = solara.use_reactive(open)  # type: ignore
+    style_flat = solara.util._flatten_style(style)
+
+    def set_dates_cast(values):
+        date_value = cast(
+            Tuple[Optional[dt.date], Optional[dt.date]], tuple([dt.datetime.strptime(item, "%Y-%m-%d").date() if item is not None else None for item in values])
+        )
+        value.value = date_value
+
+    input = solara.v.TextField(
+        label=label,
+        value=" - ".join(date_strings),
+        on_v_model=set_dates_cast,
+        append_icon="mdi-calendar",
+        readonly=True,
```

Comment:
I think we want to remove this right, given that we respond to changes (on_v_model)?